src/main/kotlin/no/digdir/service_catalog/service/PublicServiceService.kt

```diff
@@ -1,10 +1,17 @@
 package no.digdir.service_catalog.service
 
+import no.digdir.service_catalog.model.PublicService
 import no.digdir.service_catalog.mongodb.PublicServiceRepository
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 
 @Service
 class PublicServiceService(private val publicServiceRepository: PublicServiceRepository) {
     fun findPublicServicesByCatalogId(catalogId: String) =
         publicServiceRepository.getByCatalogId(catalogId)
+
+    fun findPublicServiceById(id: String, catalogId: String) =
+        publicServiceRepository.findByIdOrNull(id)?.takeIf {
+            it.catalogId == catalogId
+        }
```

Comment:
og flytt takeIf ned på en ny linje, det blir vanskeligere å lese koden når du kjører to metoder på samme linje
```suggestion
        publicServiceRepository.findByIdOrNull(id)
            ?.takeIf { it.catalogId == catalogId }
```

Reply:
Hvis du føler at du må ha innholdet i takeIf på en egen linje så kan du gjøre det etterpå
```suggestion
        publicServiceRepository.findByIdOrNull(id)
            ?.takeIf {
                it.catalogId == catalogId
            }
```